core/src/main/java/org/projectnessie/cel/extension/StringsLib.java

```diff
@@ -297,14 +422,147 @@ static String replace(String str, String old, String replacement) {
     return str.replace(old, replacement);
   }
 
+  /**
+   * replace first n non-overlapping instance of {old} replaced by {replacement}. It works as
+   * https://pkg.go.dev/strings#Replace to have consistent behavior as cel in Go
+   *
+   * <p>if n == 0, there is no change to the string
+   *
+   * <p>if n < 0, there is no limit on the number of replacement
+   *
+   * <p>if {old} is empty, it matches at the beginning of the string and after each UTF-8 sequence,
+   * yielding up to k+1 replacements for a k-rune string
+   *
+   * @param str
+   * @param old
+   * @param replacement
+   * @param n
+   * @return
+   */
+  static String replaceN(String str, String old, String replacement, int n) {
+    if (n == 0 || old.equals(replacement)) {
+      return str;
+    }
+    if (n < 0) {
+      return str.replace(old, replacement);
+    }
+    char[] chars = str.toCharArray();
```

Comment:
Looks like you can get rid of this array by using `StringBuilder.append()` taking the `String` 
```java
    StringBuilder stringBuilder = new StringBuilder();
    int index = 0;
    int count = 0;

    for (; count < n && index < str.length(); count++) {
      if (old.length() == 0) {
        stringBuilder.append(replacement).append(str, index, index + 1);
        index++;
      } else {
        int found = str.indexOf(old, index);
        if (found == -1) {
          // not found, append to the end
          stringBuilder.append(str, index, str.length());
          return stringBuilder.toString();
        }
        if (found > index) {
          stringBuilder.append(str, index, found);
        }
        stringBuilder.append(replacement);
        index = found + old.length();
      }
    }
    if (index < str.length()) {
      stringBuilder.append(str, index, str.length());
    }

```